src/jira/client/axios.ts

```diff
@@ -55,51 +54,49 @@ function getAuthMiddleware(secret: string) {
 	);
 }
 
-export const JiraErrorMessages = {
-	400: (detailMessage) => `HTTP 400 - Request had incorrect format. Details: ${detailMessage}`,
-	401: () => "HTTP 401 - Missing a JWT token, or token is invalid.",
-	403: () => "HTTP 403 - The JWT token used does not correspond to an app that defines the jiraDevelopmentTool module, or the app does not define the 'WRITE' scope",
-	413: () => "HTTP 413 - Data is too large. Submit fewer devinfo entities in each payload.",
-	429: () => "HTTP 429 - API rate limit has been exceeded.",
-	"default": (status, detailMessage) => `HTTP ${status} - ${detailMessage}`
+export const getJiraErrorMessages = (status:number, message?:string) => {
+	switch(status) {
+		case 400: return `HTTP 400 - Request had incorrect format. Details: ${message}`;
+		case 401: return "HTTP 401 - Missing a JWT token, or token is invalid.";
+		case 403: return "HTTP 403 - The JWT token used does not correspond to an app that defines the jiraDevelopmentTool module, or the app does not define the 'WRITE' scope";
+		case 413: return "HTTP 413 - Data is too large. Submit fewer devinfo entities in each payload.";
+		case 429: return "HTTP 429 - API rate limit has been exceeded.";
+		default: return `HTTP ${status} - ${message}`;
+	}
 };
 
 /**
  * Middleware to enhance failed requests in Jira.
  */
-function getErrorMiddleware() {
+function getErrorMiddleware(logger: Logger) {
 	return (
 		/**
 		 * Potentially enrich the promise's rejection.
 		 *
 		 * @param {import("axios").AxiosError} error - The error response from Axios
 		 * @returns {Promise<Error>} The rejected promise
 		 */
-		(error) => {
-			if (error.response) {
-				const { status } = error.response || {};
+		(error: AxiosError): Promise<Error> => {
+			if (error?.response) {
+				const status = error.response.status;
 
 				// truncating the detail message returned from Jira to 200 characters
 				const detailMessage = JSON.stringify(error?.response?.data)?.substring(0, 200);
+				const errorMessage = getJiraErrorMessages(status, detailMessage);
 
-				const errorMessage = status in JiraErrorMessages
-					? JiraErrorMessages[status](detailMessage)
-					: JiraErrorMessages["default"](status, detailMessage)
-
-				return Promise.reject(new JiraClientError(error, errorMessage));
-			} else {
-				return Promise.reject(error);
+				error.message = `Jira Client Error: ${errorMessage}`;
 			}
-		}
-	);
+			logger.error(error);
```

Comment:
1. Would It cause double-logging if we log this error on the top level as well?
2. Do we log it in case of 401s and 404s and other non-erroneous conditions? 

 [ ] Make sure no double logging occur here, change to warn if it includes 401 and 404s

Reply:
What do you mean at the top level?  Because of the promise rejection from axios?  The only reason it used to capture before is because we used to just throw an error.  This is a bit cleaner. 

But you're right about 400s.  I should check to see what's the status and log according to that.